.github/workflows/publish.yml

```diff
@@ -0,0 +1,10 @@
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
```

Comment:
and i guess along with this we should set up a branch protection rule for this tag pattern, right? or a tag protection rule, or whatever they're called, that only allows the maintainers of this repo to push those tags.

Reply:
Yeah I just went ahead and added a wildcarded (`*`) protection rule which should prohibit tag creation except by admins or other folks with org-wide elevated perms.